modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-partner-portal/extra/remote-app/src/routes/MDFClaimForm/components/MDFClaimPage/schema/yup.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {array, mixed, number, object, string} from 'yup';
+
+const permitedDocument = {
+	maxSize: 100000,
+	types: [
+		'image/jpg',
+		'image/jpeg',
+		'image/gif',
+		'image/png',
+		'application/pdf',
+	],
+};
+
+const claimSchema = object({
+	activities: array()
+		.test(
+			'needAtLeatOneSelectedActivity',
+			'Need at least one selected activity',
+			(activities) =>
+				activities?.some((activity) => activity.selected) ? true : false
+		)
+		.test(
+			'needMoreThanOneSelectedActivity',
+			'Need at least one invoice uploaded',
+			(activities) =>
+				activities?.some(
+					(activity) =>
+						activity.selected &&
+						activity.budgets.some((budget: any) =>
+							budget.invoice ? true : false
+						)
+				)
+					? true
+					: false
+		)
```

Comment:
acredito que da pra fazer todas as validacoes sem o test. So precisa olhar os outros yups presente. 

`array().min(1, 'Required').max(3, 'You have exceed the choose limit'),`

Reply:
Antes de mandar a PR, importante ver a doc da lib com calma.

https://github.com/jquense/yup

Tem bastante coisa que ja tem nativa e nao precisa usar nenhum loop pra fazer.

Reply:
Eu tentei a chegar utilizar das funções nativas do yup min e max, mas ela não funciona para o tipo de teste que queriamos validar, eu ate tentei e debuguei no console do error do formikHelpers, mas ele nao seta a mensagem de erro min e maximo, não põe no objeto do formik, de primeira também fui na expectativa de dar certo com elas, mas não funcionou e ai eu tentei por uma terceira via.